zipline/data/history_loader.py

```diff
@@ -31,83 +31,30 @@
 from zipline.utils.numpy_utils import float64_dtype
 
 
-class SlidingWindow(object):
-    """
-    Wrapper around an AdjustedArrayWindow which supports monotonically
-    increasing (by datetime) requests for a sized window of data.
+class HistoryCompatibleUSEquityAdjustmentReader(object):
 
-    Parameters
-    ----------
-    window : AdjustedArrayWindow
-       Window of pricing data with prefetched values beyond the current
-       simulation dt.
-    cal_start : int
-       Index in the overall calendar at which the window starts.
-    """
-
-    def __init__(self, window, size, cal_start, offset):
-        self.window = window
-        self.cal_start = cal_start
-        self.current = around(next(window), 3)
-        self.offset = offset
-        self.most_recent_ix = self.cal_start + size
+    def __init__(self, adjustment_reader):
+        self._adjustments_reader = adjustment_reader
 
-    def get(self, end_ix):
+    def load_adjustments(self, columns, dts, assets, output_offset=0):
```

Comment:
@ssanderson do you have a better alternative than `output_offset` for this value which changes the values so that `AdjustedArrayWindow` pops the adjustment an index earlier? e.g. `{2: Float64Window(0, 2, 0, 0, 0.5)`} becomes `{1: Float64Window(0, 2, 0, 0, 0.5)`